src/guess.py

```diff
@@ -202,30 +204,34 @@ def bayesdb_guess_stattypes(column_names, rows, null_values=None,
     for ci, column_name in enumerate(column_names):
         if casefold(column_name) in override_map:
             stattype = override_map[casefold(column_name)]
+            reason = 'User override.'
         else:
             column = nullify(null_values, rows, ci)
-            stattype = guess_column_stattype(column,
+            [stattype, reason] = guess_column_stattype(column,
                                              distinct_ratio=distinct_ratio,
                                              nullify_ratio=nullify_ratio,
                                              numcat_count=numcat_count,
                                              numcat_ratio=numcat_ratio,
                                              have_key=(key is not None))
             if stattype == 'key':
                 key = column_name
-        stattypes.append(stattype)
+        stattypes.append([stattype, reason])
     return stattypes
 
-def guess_column_stattype(column, **kwargs):
+def guess_column_stattype(column, reason='', **kwargs):
     counts = count_values(column)
     if None in counts:
         del counts[None]
     if len(counts) < 2:
-        return 'ignore'
+        return ['ignore', '%s There is only one unique value.' %(reason,)]
     (most_numerous_key, most_numerous_count) = sorted(
         counts.items(), key=lambda item: item[1], reverse=True)[0]
     if most_numerous_count / float(len(column)) > kwargs['nullify_ratio']:
         column = [ None if v == most_numerous_key else v for v in column ]
-        return guess_column_stattype(column, **kwargs)
+        return guess_column_stattype(column, '%s More than %d percent of the '
```

Comment:
I see, `reason` is used in the recursive call, although I am confused about this behavior.